### PR TITLE
Add RFC for deprecation Ember Data initializers

### PR DESCRIPTION
**Pull request by bmac**
[Rendered](https://github.com/bmac/rfcs/blob/0c24fab47abd0f5dd173f6273223f6c5236e073a/text/0181-deprecate-ember-data-initializers.md).

## Known usage (checked for PR sent)

- [x] bc-ember-data-sails [PR](https://github.com/brickclick/ember-data-sails/pull/1) 
  - ./bc-ember-data-sails/addon/initializers/ember-data-sails.js:  before: 'store',
  - Last updated one year ago.
- [x] ember-data-complex [PR](https://github.com/foxnewsnetwork/ember-data-complex/pull/1)
  - ./ember-data-complex/addon/initializers/data-complex.coffee:  before: 'store'
  - Last updated two years ago
- [x] ember-data-model-fragments [PR](https://github.com/lytics/ember-data-model-fragments/pull/226)
  - ./ember-data-model-fragments/app/initializers/model-fragments.js:  before: "store",
- [x] ember-data-sails [PR](https://github.com/huafu/ember-data-sails/pull/39)
  - ./ember-data-sails/addon/initializers/ember-data-sails.js:  before: 'store',
- [x] ember-fryctoria [PR](https://github.com/poetic/ember-fryctoria/pull/4)
  - ./ember-fryctoria/addon/initializers/extend-ds-model.js:  before:     'store',
  - ./ember-fryctoria/addon/initializers/syncer.js:  before:     'store',
  - ./ember-parse/addon/initializers/parse.js:  before: 'store',
- [x] ember-solr [PR](https://github.com/chriseldredge/ember-solr/pull/1)
  - ./ember-solr/addon/initializers/solr.js:  before: 'store',
  - Last updated two years ago
- [x] ember-youtube-data-model [PR](https://github.com/foxnewsnetwork/ember-youtube-data-model/pull/1)
  - ./ember-youtube-data-model/addon/initializers/youtube.coffee:  before: 'store'
  - Last updated two years ago

## Todo

- RFC - [PR](https://github.com/emberjs/data/pull/4657)
  - [x] Submitted
  - [x] FCP
  - [ ] Merged (pending FCP)
- Deprecation Guide - [PR](https://github.com/emberjs/website/pull/2781)
  - [x] Submitted
  - [x] Merged
  - [ ] Updated (pending RFC)
- Code change - [PR](https://github.com/emberjs/data/pull/4657)
  - [x] Submitted
  - [ ] Merged (pending RFC)

### CODE REVIEW

File: text/0181-deprecate-ember-data-initializers.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+The goal of this RFC is to remove the `data-adapter`, `injectStore`,
+`transforms`, and `store` Ember initializers that Ember Data injects
+into apps.  The `ember-data` initializer not be changed and any code
+that previously depended on the ordering of these initializers (via
+the `before` or `after` properties on an initalizer) will can be
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: "will can be changed"

File: text/0181-deprecate-ember-data-initializers.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+- Start Date: 2016-11-22
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+The goal of this RFC is to remove the `data-adapter`, `injectStore`,
+`transforms`, and `store` Ember initializers that Ember Data injects
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: possibly better to refer to them as application/application instance initializers?

File: text/0181-deprecate-ember-data-initializers.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+into apps.  The `ember-data` initializer not be changed and any code
+that previously depended on the ordering of these initializers (via
+the `before` or `after` properties on an initalizer) will can be
+changed to use the `ember-data` initializers for ordering.
+
+# Motivation
+
+The intializes `data-adapter`, `injectStore`, `transforms`, and
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: initializers

File: text/0181-deprecate-ember-data-initializers.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+`store` have not been used by Ember Data since
+[Apr 8, 2014](https://github.com/emberjs/data/commit/d25e23f622a3677b8372db535b2ab824ad306a16). However,
+they are still injected into every Ember app that depends on Ember
+Data because existing apps may have depending on these initializers
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: "may have depending"

File: text/0181-deprecate-ember-data-initializers.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+for ordering their own initializers to run before or after Ember
+Data's setup code.
+
+Removing these initializers will help reduce the amount of code Ember
+Data needs to support.
+
+Since these initializers are noop functions that run after the
+`ember-data` initializer. Any initializers that depends on one of the
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: comma instead of period?

File: text/0181-deprecate-ember-data-initializers.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+deprecated initializers listed in this rfc can easly be replaced by
+depending on the `ember-data` initializer instead.
+
+# Detailed design
+
+Ember Data's instance initializer will start checking for any
+initializers who's `before` or `after` properties depend on one of
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: whose

File: text/0181-deprecate-ember-data-initializers.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+these deprecated initalizer. If it finds an initalizer that references
+one of the deprecated initalizers Ember Data will then log a
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: initializers,

File: text/0181-deprecate-ember-data-initializers.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+This deprecation message will continue to appear until Ember Data
+3.0.0 when these initalizers and the deprecation code will be finally
+removed.
+
+
+# How We Teach This
+
+This change should have no impact on how we teach Ember or Ember
+Data. The initalizers that will be remove have been unused for a long
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: "will be remove"

File: text/0181-deprecate-ember-data-initializers.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+and suggest changing the `before` or `after` property (the deprecation
+message will refer to the correct property dynamically) to depend on
+Ember Data instead.
+
+This deprecation message will continue to appear until Ember Data
+3.0.0 when these initalizers and the deprecation code will be finally
+removed.
+
+
+# How We Teach This
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: could we come up with a codemod?

File: text/0181-deprecate-ember-data-initializers.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+This change should have no impact on how we teach Ember or Ember
+Data. The initalizers that will be remove have been unused for a long
+time and are not mentioned anywhere in today's guides or API docs.
+
+Users who need to run initalizer code before or after Ember Data
+injects the store into routes should be taught to use `before:
+'ember-data'`, or `after: 'ember-data'` on there initalizers.
```
Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: their

Comment (locks) [inline text/0181-deprecate-ember-data-initializers.md]: initializers

**Comment by locks** [comment]
@bmac as done in https://github.com/cibernox/rfcs/blob/deprecate-ember-k/text/0000-deprecate-ember-k.md#addenda, I would like to suggestion reaching out to EmberObserver and get some usage statistics across published addons (worried about outdated dependencies), and coming up with a codemod, if possible. Looking good!

File: text/0181-deprecate-ember-data-initializers.md (hunk 8 above)
Comment (tragiclifestories) [inline text/0181-deprecate-ember-data-initializers.md]: Muphry's law kicking in here ... "removed" :trollface:

**Comment by bmac** [comment]
@locks I checked out all of the [Ember Data Adapters](https://emberobserver.com/categories/ember-data-adapters) and [Ember Data Extensions](https://emberobserver.com/categories/ember-data-extensions) form Ember observer and found the following adapters referencing the deprecated initializers.

```
./bc-ember-data-sails/addon/initializers/ember-data-sails.js:  before: 'store',
./ember-data-complex/addon/initializers/data-complex.coffee:  before: 'store'
./ember-data-model-fragments/app/initializers/model-fragments.js:  before: "store",
./ember-data-sails/addon/initializers/ember-data-sails.js:  before: 'store',
./ember-fryctoria/addon/initializers/extend-ds-model.js:  before:     'store',
./ember-fryctoria/addon/initializers/syncer.js:  before:     'store',
./ember-parse/addon/initializers/parse.js:  before: 'store',
./ember-solr/addon/initializers/solr.js:  before: 'store',
./ember-youtube-data-model/addon/initializers/youtube.coffee:  before: 'store'
```

**Comment by locks** [comment]
The Ember Data team has reviewed this proposal and have decided to move it to the Final Comment Period! If no new significant concerns are raised, in one week's time this proposal will be officially merged.
Thanks everyone for their work!

**Pull request merged**